lighthouse-extension/app/src/popup.js

```diff
@@ -117,18 +117,27 @@ function createOptionItem(text, id, isChecked) {
  * @param {!Window} background Reference to the extension's background page.
  * @param {{selectedCategories: !Object<boolean>, useDevTools: boolean}} settings
  */
-function onGenerateReportButtonClick(background, settings) {
+async function onGenerateReportButtonClick(background, settings) {
   showRunningSubpage();
 
   const feedbackEl = document.querySelector('.feedback');
   feedbackEl.textContent = '';
 
   const {selectedCategories, useDevTools} = settings;
 
-  background.runLighthouseInExtension({
-    restoreCleanState: true,
-    flags: {throttlingMethod: useDevTools ? 'devtools' : 'simulate'},
-  }, selectedCategories).catch(err => {
+  try {
+    await background.runLighthouseInExtension({
+      restoreCleanState: true,
+      flags: {throttlingMethod: useDevTools ? 'devtools' : 'simulate'},
+    }, selectedCategories);
+
+    // Close popup once report is opened in a new tab
+    window.close();
+  } catch (err) {
+    handleError(err);
+  }
+
+  function handleError(err) {
```

Comment:
any reason to make this a function? I'd say either spin off as it's own function or just nest in the catch{}